GVFS/GVFS.Platform.Mac/MacFileBasedLock.cs

```diff
@@ -0,0 +1,159 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace GVFS.Platform.Mac
+{
+    public class MacFileBasedLock : FileBasedLock
+    {
+        // #define O_WRONLY    0x0001      /* open for writing only */
+        private const int OpenWriteOnly = 0x0001; 
+
+        // #define O_CREAT     0x0200      /* create if nonexistant */
+        private const int OpenCreate = 0x0200;    
+
+        // #define EINTR       4       /* Interrupted system call */
+        private const int EIntr = 4; 
+
+        // #define EAGAIN      35      /* Resource temporarily unavailable */
+        // #define EWOULDBLOCK EAGAIN  /* Operation would block */
+        private const int EWouldBlock = 35; 
+
+        private const int InvalidFileDescriptor = -1;
+
+        private static readonly ushort FileMode644 = Convert.ToUInt16("644", 8);
+
+        private int lockFileDescriptor;
+
+        public MacFileBasedLock(
+            PhysicalFileSystem fileSystem,
+            ITracer tracer,
+            string lockPath,
+            string signature)
+            : base(fileSystem, tracer, lockPath, signature) 
+        {
+            this.lockFileDescriptor = InvalidFileDescriptor;
+        }
+
+        [Flags]
+        private enum FLockOperations

```

Comment:
Also a nit: you're currently using constants for some and an enum for others. I would keep them consistent, and the constants seem to make the usage less cluttered (no need for a cast)

Reply:
I'll make them all contants